pkg/lexer/state.go

```diff
@@ -52,12 +62,14 @@ next:
 			break
 		}
 
+		// commands don't start with a keyword
 		l.emit(cmdOpLookup(word))
 		fallthrough
 	default:
 		lexCmd(l)
 	}
 
+	// loop till eof
```

Comment:
I think this project is overusing the `goto` keyword 😂 There is a reason for having loop keywords... but conceptually they do the same thing so, I guess its fine.

Honestly I don't know the answer to this but the main reason I avoid goto is precisely the weird control flow that we have in some of the very low level code in linux kernel and makes it difficult to reason about in large code bases.

Up to you to keep it but I think its clear that `for r := l.peek(); r != eof; {...}` does the same. And compilers are optimised enough to work with the `for` loops as efficiently as `goto` statements.
Second I argue that it is easier to read code with `for` than `goto` - another reason why the loop keywords exist...

Reply:
I replaced most of the `goto` statements with `for` loops.